syft/pkg/cataloger/golang/scan_bin.go

```diff
@@ -26,20 +25,23 @@ func scanFile(reader unionReader, filename string) ([]*debug.BuildInfo, []string
 	// with more than one binary
 	readers, err := getReaders(reader)
 	if err != nil {
-		log.Warnf("golang cataloger: opening binary: %v", err)
+		log.Warnf("golang cataloger: failed to open a binary: %v", err)
 		return nil, nil
 	}
 
 	var builds []*debug.BuildInfo
 	for _, r := range readers {
 		bi, err := buildinfo.Read(r)
-		if errors.As(err, &errNotGoExe) {
-			log.Debugf("golang cataloger: scanning file %s: %v", filename, err)
-			return nil, nil
-		}
 
-		if err != nil {
-			log.Warnf("golang cataloger: scanning file %s: %v", filename, err)
+		// note: the stdlib does not export the error we need to check for
+		if err.Error() == "not a Go executable" {
```

Comment:
Is this matching some text from the go lib?

Reply:
indeed, slightly more context here: https://github.com/anchore/syft/pull/904#discussion_r832209830